api_tests/nodes/views/test_node_reorder_components.py

```diff
@@ -0,0 +1,119 @@
+    project.add_contributor(read_contrib, permissions.READ)
+    project.save()
+    return project
+
+@pytest.fixture()
+def url(project):
+    return f'/{API_BASE}nodes/{project._id}/reorder_components/'
+
+
+@pytest.mark.django_db
+class TestNodeReorderComponents:
+
+    @pytest.fixture()
+    def children(self, project, admin_contrib):
+        child1 = NodeFactory(parent=project, is_public=False, creator=admin_contrib)
+        child2 = NodeFactory(parent=project, is_public=False, creator=admin_contrib)
+        child3 = NodeFactory(parent=project, is_public=True, creator=admin_contrib)
+
+        rel1 = NodeRelation.objects.get(parent=project, child=child1)
+        rel2 = NodeRelation.objects.get(parent=project, child=child2)
+        rel3 = NodeRelation.objects.get(parent=project, child=child3)
+        project.set_noderelation_order([rel2.pk, rel3.pk, rel1.pk])
+
+        return [child1, child2, child3]
+
+    @pytest.fixture
+    def payload_asc(self, children):
+        return {
+            'data': [
+                {
+                    'type': 'nodes',
+                    'id': each._id,
+                    'attributes': {
+                        '_order': pos
+                    }
+                } for pos, each in enumerate(children)
+            ]
+        }
+
+    @pytest.fixture
+    def payload_desc(self, children):
+        return {
+            'data': [
+                {
+                    'type': 'nodes',
+                    'id': each._id,
+                    'attributes': {
+                        '_order': pos
+                    }
+                } for pos, each in enumerate(children[::-1])
+            ]
+        }
+
+    def test_reorder_components_admin(
+            self, app, project, admin_contrib, children, url, payload_asc, payload_desc
+    ):
+        res = app.patch_json_api(url, payload_asc, auth=admin_contrib.auth, expect_errors=True, bulk=True)
+        project.reload()
+        component_order = [el.child._id for el in project.node_relations.all()]
+        assert res.status_code == 200
+        assert component_order == [each._id for each in children]
+
+        res = app.put_json_api(url, payload_desc, auth=admin_contrib.auth, expect_errors=True, bulk=True)
+        project.reload()
+        component_order = [el.child._id for el in project.node_relations.all()]
+        assert res.status_code == 200
+        assert component_order == [each._id for each in children[::-1]]
+
+    def test_reorder_components_write_read(
+            self, app, write_contrib, read_contrib, url, payload_asc
+    ):
+        res = app.patch_json_api(url, payload_asc, auth=write_contrib.auth, expect_errors=True, bulk=True)
+        assert res.status_code == 403
+
+        res = app.patch_json_api(url, payload_asc, auth=read_contrib.auth, expect_errors=True, bulk=True)
+        assert res.status_code == 403
+
+    def test_reorder_invalid_component(
```

Comment:
There should be more test cases in case of invalid payload and conflicting instructions, including the following:

- _order is higher the list is long (for one and multiple payload items)
- _order is lower then zero long (for one and multiple payload items)
- two or more items in the payload have the same _order
- an item id appears in the payload >1 time with the same _order information
 - an item id appears in the payload >1 time with the different _order information

That a summary of the types of validation cases I'd like to have descriptive error messages for, it doesn't need a unique message for every one, but it should distinguish 400s Invalid Payload from 409 Conflicting Information and let the develop know the nature of the mistake they've made.